ragaai_catalyst/tracers/tracer.py

```
@@ -516,41 +516,8 @@ def set_dataset_name(self, dataset_name):
-            'interval_time': self.interval_time,
-            'max_upload_workers': self.max_upload_workers,
-            'external_id': self.external_id
-        }
-            
-            # Reinitialize self with new dataset_name and stored parameters
-            self.__init__(
-                dataset_name=dataset_name,
-                **current_params
-            )
+        self.dynamic_exporter.dataset_name = dataset_name
+        logger.debug(f"Updated dynamic exporter's dataset_name to {dataset_name}")
```

Pylint (3.3.7)

[warning] Use lazy % formatting in logging functions

(W1203)